### PR TITLE
Ruby on Rails folder: Updated README.md

### LINKED ISSUE #24034: Ruby on Rails folder: update README.md
**1. Description:**

READMEs need to be updated throughout the curriculum repository.

Related to: 
- https://github.com/TheOdinProject/top-meta/issues/109

### Course README template:

```markdown
# <course_name>

/* in case course belongs to multiple paths, make sure to link those paths */ 

This folder contains lesson markdown files that make up the <course_name> course. This course exists in the [<path_name>](<path_link>) path on the Odin Project Website.

## Course Outline

The following list represents how the lessons are divided into sections and presented on the website.

**Disclaimer:** Given the ever updating nature of the curriculum, the outline might be outdated. See the [<course_name> course on the website](<course_link>)
instead.

### <section_name>

/* make sure the lesson order corresponds to the order on the website */
/* make sure to use relative links with respect to this README to link the markdown file */

1. [<lesson_name>](<lesson_markdown_file_link>)
2. [**<project_name>**](<project_markdown_file_link>)
```

**2. Acceptance Criteria:**

- [ ] Update Ruby on Rails folder's `README.md` file to the template provided above
- [ ] Remove comments from the template
- [ ] Fill in the appropriate information in the angle (`<>`) brackets in the template
  - Make sure to the path/course/lesson names correspond to what is on the website  
- [ ] Use relative links to link the markdown files
  - For example use `apis/project_kittens_apis.md` as the link to the markdown file instead of `https://github.com/TheOdinProject/curriculum/blob/main/ruby_on_rails/apis/project_kittens_api.md` 
  - Make sure the links work as intended

**3. Additional Information:**

If anyone wants to work on this, please comment to be assigned. In case of any questions, feel free to ask below.

### PR DESCRIPTION
**Pull request by nguyens20**
Updated the README for Ruby on Rails to follow the new template.

Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [ ] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [ ] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

Closes #24034 

**2. This PR:**
 
Updated the README.md document to follow the provided template.
Fixed broken links.
Made projects bold.
Adjusted README.md layout so that it the lessons order matches that of the website.

**3. Additional Information:**

### CODE REVIEW

File: ruby_on_rails/README.md [hunk 1]
```diff
@@ -1,45 +1,56 @@
...
+
+### Introduction
+1. [How this Course Will Work](introduction/how_this_course_will_work.md)
+2. [Preparing for Deployment](introduction/preparing_for_deployment.md)
+3. [**Project: Installing Rails**](introduction/project_installing_rails.md)
+4. [A Railsy Web Refresher](introduction/web_refresher.md)
+### Rails Basics
+1. [Routing](rails_basics/routing.md)
+2. [Controllers](rails_basics/controller_basics.md)
+3. [Views](rails_basics/views.md)
```
Comment (01zulfi) [inline ruby_on_rails/README.md:18]: The markdown file link for the Controllers lesson is broken. It needs to be fixed.

**Review by 01zulfi — CHANGES_REQUESTED** [review]

File: ruby_on_rails/README.md (hunk 1 above)
Comment (nguyens20) [reply ruby_on_rails/README.md]: I made a new commit to fix the issue with the Controllers lesson markdown file link. Please let me know if there is anything else to add/fix !

File: ruby_on_rails/README.md [hunk 2]
```diff
@@ -1,45 +1,56 @@
...
+**Disclaimer:** Given the ever updating nature of the curriculum, the outline might be outdated. See the [Ruby on Rails course on the website](https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails) instead.
+
+### Introduction
+1. [How this Course Will Work](introduction/how_this_course_will_work.md)
+2. [Preparing for Deployment](introduction/preparing_for_deployment.md)
+3. [**Project: Installing Rails**](introduction/project_installing_rails.md)
+4. [A Railsy Web Refresher](introduction/web_refresher.md)
+### Rails Basics
+1. [Routing](rails_basics/routing.md)
+2. [Controllers](rails_basics/controller.md)
```
Comment (01zulfi) [inline ruby_on_rails/README.md:17]:
```suggestion
2. [Controllers](rails_basics/controllers.md)
```

**Review by 01zulfi — CHANGES_REQUESTED** [review]

**Review by 01zulfi — APPROVED** [review]
Looks great. Thanks for working on this! :fire:

**Pull request merged**